### PR TITLE
Migrate appRouter to ES6

### PR DESCRIPTION
**Pull request by MrTimscampi**
**Changes**

This is a large one.

It converts `appRouter` to a class (`AppRouter`) and brings back the overrides from `site.js` into the class definition (removing a few duplicates).

**Issues**

### CODE REVIEW

**Comment by Camc314** [comment]
Playback shows no OSD, and the nowplaying page is broken :p

File: src/controllers/playback/video/index.js [hunk 1]
```diff
@@ -1442,7 +1440,7 @@ import 'css!assets/css/videoosd';
                 });
             } catch (e) {
                 import('appRouter').then(({default: appRouter}) => {
-                    appRouter.goHome();
+                    appRouter.default.goHome();
```
Comment (Camc314) [inline src/controllers/playback/video/index.js:1443]:
```suggestion
                    appRouter.goHome();
```

File: src/components/appRouter.js [hunk 2]
```diff
@@ -1,66 +1,298 @@
...
+                this.showSelectServer();
                 break;
             case 'ConnectSignIn':
-                appRouter.showWelcome();
+                this.showWelcome();
                 break;
             case 'ServerUpdateNeeded':
-                require(['alert'], function (alert) {
-                    alert.default({
+                require(['alert'], (alert) => {
```
Comment (Camc314) [inline src/components/appRouter.js:290]:
```suggestion
                import('alert').then(({default: alert}) =>{
```

File: src/components/appRouter.js [hunk 3]
```diff
@@ -83,27 +315,27 @@ define(['loading', 'globalize', 'events', 'viewManager', 'skinManager', 'backdro
...
+            url = this.baseUrl() + url;
         }
 
         if (ctx.querystring && route.enableContentQueryString) {
             url += '?' + ctx.querystring;
         }
 
-        require(['text!' + url], function (html) {
-            loadContent(ctx, route, html, request);
+        require(['text!' + url], (html) => {
```
Comment (Camc314) [inline src/components/appRouter.js:325]:
```suggestion
        import('text!' + url).then(({default: html}) => {
```

File: src/components/appRouter.js [hunk 4]
```diff
@@ -158,64 +389,62 @@ define(['loading', 'globalize', 'events', 'viewManager', 'skinManager', 'backdro
...
-    function onForcedLogoutMessageTimeout() {
-        var msg = forcedLogoutMsg;
-        forcedLogoutMsg = null;
+    onForcedLogoutMessageTimeout() {
+        const msg = this.forcedLogoutMsg;
+        this.forcedLogoutMsg = null;
 
         if (msg) {
-            require(['alert'], function (alert) {
+            require(['alert'], (alert) => {
```
Comment (Camc314) [inline src/components/appRouter.js:409]:
```suggestion
            import('alert').then(({default: alert}) => {
```

File: src/scripts/site.js [hunk 5]
```diff
@@ -653,7 +654,9 @@ function initClient() {
             nowPlayingHelper: componentsPath + '/playback/nowplayinghelper',
             pluginManager: componentsPath + '/pluginManager',
             packageManager: componentsPath + '/packageManager',
-            screensaverManager: componentsPath + '/screensavermanager'
+            screensaverManager: componentsPath + '/screensavermanager',
+            chromecastHelper: 'plugins/chromecastPlayer/chromecastHelpers',
```
Comment (dkanada) [inline src/scripts/site.js:658]: I moved this to the plugin so it's all self contained.

Comment (MrTimscampi) [reply src/scripts/site.js:658]: This might be a rebasing artifact. Will fix.

**Comment by nyanmisaka** [comment]
@MrTimscampi 
Playback settings in the video OSD seem to be broken since this PR. Once I click on one of the options, the playback will exit and jump to the homepage. Tried on edge, chrome and firefox.

settings

File: src/components/dialogHelper/dialogHelper.js [hunk 6]
```diff
@@ -375,7 +375,7 @@ import 'scrollStyles';
             dlg.setAttribute('data-lockscroll', 'true');
         }
 
-        if (options.enableHistory !== false && appRouter.enableNativeHistory()) {
+        if (options.enableHistory !== false) {
```
Comment (dmitrylyzo) [inline src/components/dialogHelper/dialogHelper.js:378]: This broke menu (see @nyanmisaka post). Dialogs+history is too fragile combination

Comment (nyanmisaka) [reply src/components/dialogHelper/dialogHelper.js:378]: revert `appRouter.enableNativeHistory()` indeed fix that.

**Pull request merged**